06-building-a-program-start-to-finish/writing-tests/Output/test/spec/inverted-index.spec.js

```diff
@@ -0,0 +1,153 @@
+
+describe('Normalized Text', () => {
+  it('should return an array containing only alphabets', () => {
+    expect(newIndex.normalizedText(sampleData)).to.not.include(' ');
+    expect(newIndex.normalizedText(sampleData)).to.match(/^[a-zA-Z]/);
+  });
+
+  it('should return an array containing the correct number of words', () => {
+    expect(newIndex.normalizedText(sampleData).length).to.equal(10);
+  });
+
+  it('should return an array containing the correct words', () => {
+    expect(newIndex.normalizedText(sampleData)).to.eql(['mary',
+      'had',
+      'a',
+      'little',
+      'lamb',
+      'a',
+      'little',
+      'lamb',
+      'mary',
+      'had']);
+  });
+});
+
+describe('Unique words', () => {
+  it('should not return any duplicate words', () => {
+    expect(newIndex.uniqueWords(sampleData).length).to.equal(5);
+    expect(newIndex.uniqueWords(sampleData))
+      .to.eql(['mary', 'had', 'a', 'little', 'lamb']);
+  });
+});
+
+describe('Populate index', () => {
+  newIndex.createIndex('books.json', books);
+  newIndex.createIndex('books2.json', books2);
+  it('should create index once JSON file has been read', () => {
+    expect(Object.keys(newIndex.indices).length).to.equal(2);
+  });
+
+  it('should map words to the correct document location', () => {
+    expect(newIndex.getIndex('books.json')).to.eql({
+      alice: [0],
+      in: [0, 1, 2],
+      wonderland: [0],
+      falls: [0],
+      into: [0],
+      a: [0, 1, 2],
+      rabbit: [0],
+      hole: [0],
+      and: [0, 1, 2],
+      enters: [0],
+      world: [0],
+      full: [0],
+      of: [0, 1, 2],
+      imagination: [0],
+      the: [1, 2],
+      lord: [1, 2],
+      rings: [1, 2],
+      fellowship: [1, 2],
+      ring: [1, 2],
+      an: [0, 1, 2],
+      unusual: [1, 2],
+      alliance: [1, 2],
+      man: [1, 2],
+      elf: [1, 2],
+      dwarf: [1, 2],
+      wizard: [1, 2],
+      hobbit: [1, 2],
+      seek: [1, 2],
+      to: [0, 1, 2],
+      destroy: [1, 2],
+      powerful: [1, 2]
+    });
+  });
+});
+
+describe('Get index', () => {
+  it('should return the accurate index for indexed files', () => {
+    expect(newIndex.getIndex('books')).to.be.defined;
+  });
+});
+
+describe('Search index', () => {
+  it('should have search index method in the class', () => {
+    expect(InvertedIndex.searchIndex).to.be.defined;
+  });
+
+  it('Should return the correct index for words searched for', () => {
+    expect(newIndex.searchIndex('fellowship', 'books.json')).to.eql({
+      fellowship: [1, 2]
+    });
+    expect(newIndex.searchIndex('Alice', 'books.json')).to.eql({
+      alice: [0]
+    });
+    expect(newIndex.searchIndex('A', 'books.json')).to.eql({
+      a: [0, 1, 2]
+    });
+  });
+
+  it('should handle a varied number of search terms', () => {
+    expect(newIndex.searchIndex('a alice alliance', 'books.json'))
+      .to.eql({ a: [0, 1, 2], alice: [0], alliance: [1, 2] });
+  });
+});
+describe('Search all uploaded files', () => {
```

Comment:
Hmmm... I don't quite get the implementation for your search all files. I'll however reserve till I see how it works. *evil-grin*